CougSat1-Ground/src/space/cougs/ground/packetprocessing/uplinkpackets/IFJRProgramUpload.java

```diff
@@ -0,0 +1,97 @@
+package space.cougs.ground.packetprocessing.uplinkpackets;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import com.sun.jmx.snmp.Timestamp;
+
+import space.cougs.ground.packetprocessing.PacketID;
+import space.cougs.ground.utils.CRC32;
+
+public class IFJRProgramUpload extends UplinkPacket {
+
+	private String programPath;
```

Comment:
Make final, and create a constructor with this passed in, remove the setter. A new file will get a new object